provisioner/k8s/main.go

```diff
@@ -43,26 +68,74 @@ func main() {
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
 	cfg := ctrl.GetConfigOrDie()
+	kubeClient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		setupLog.Error(err, "unable to create kubernetes client")
+		os.Exit(1)
+	}
+	dependentRequirement, err := labels.NewRequirement("core.rukpak.io/owner-name", selection.Exists, nil)
+	if err != nil {
+		setupLog.Error(err, "unable to create dependent label selector for cache")
+		os.Exit(1)
+	}
+	dependentSelector := labels.NewSelector().Add(*dependentRequirement)
 	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
 		Scheme:                 scheme,
 		MetricsBindAddress:     metricsAddr,
 		Port:                   9443,
 		HealthProbeBindAddress: probeAddr,
 		LeaderElection:         enableLeaderElection,
 		LeaderElectionID:       "510f803c.olm.operatorframework.io",
+		NewCache: cache.BuilderWithOptions(cache.Options{
+			SelectorsByObject: cache.SelectorsByObject{
+				&olmv1alpha1.BundleInstance{}: {},
+				&olmv1alpha1.Bundle{}:         {},
+			},
+			DefaultSelector: cache.ObjectSelector{
+				Label: dependentSelector,
+			},
+		}),
 	})
 	if err != nil {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
 
-	if err = (&controller.BundleController{
-		Client: mgr.GetClient(),
+	// TODO: derive pod namespace from the pod that this process is running in.
+	ns := "rukpak-system"
+
+	bundleStorage := &storage.ConfigMaps{
+		Client:     mgr.GetClient(),
+		Namespace:  ns,
+		NamePrefix: "bundle-",
+	}
+
+	if err = (&controllers.BundleReconciler{
+		Client:          mgr.GetClient(),
+		KubeClient:      kubeClient,
+		Scheme:          mgr.GetScheme(),
+		PodNamespace:    ns,
+		Storage:         bundleStorage,
+		UnpackImage:     "quay.io/joelanford/kuberpak-unpack:v0.1.0",
+		CopyBundleImage: "quay.io/tflannag/olm:cpb-hacks-latest",
```

Comment:
Note: I needed to cut my own copy-bundle-util image as the OLM image we were using to copy bundle contents doesn't work with the plain bundle format, so copy bundles that don't have the metadata/annotations.yaml file present would be considered terminal, so the underlying Pod would fail and the controller would hotloop for a while. I think longer term we'll have to house our own copy-bundle tooling and make it more agnostic to bundle formats.

Reply:
Probably answered elsewhere, but how do we get the same information that `annotations.yaml`, `properties.yaml` and `dependencies.yaml` used to convey in this plain format?